extensions/amp-date-countdown/1.0/messages.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Get locale strings or undefined.
+ * @param {string} locale
+ * @return {!Array<string>|undefined}
+ */
+export function getLocaleStrings(locale) {
+  return LOCALE_WORD[locale];
+}
+
+/**
+ * Strings representing years, minutes, etc. in various locales
+ * @type {Object<string, !Array<string>>}
+ */
+const LOCALE_WORD = {
```

Comment:
Why don't we just store this in the expected `{ years: … }` format? We could then avoid the `useMemo` and everything else necessary.

Reply:
So the idea would be to do something like `LOCALE_WORD[props['locale']]` at each iteration?

One thing I am concerned about is there is a helper function that checks that the `props['locale']` value is valid and displays a warning if not.  The warning really should only be displayed one time.  So even if we were to change the format, we would still have to `useMemo` to store some info about what our `locale` is or that the warning message has already been displayed.

Happy to update the format, if you feel strongly about it.  If I am misunderstanding please let me know!